tests/testthat/test-oauth-flow-auth-code.R

```diff
@@ -90,5 +90,51 @@ test_that("forwards oauth error", {
     oauth_flow_auth_code_parse(query2, "abc")
     oauth_flow_auth_code_parse(query3, "abc")
   })
+})
+
+# can_fetch_auth_code -----------------------------------------------------
+
+test_that("external auth code sources are detected correctly", {
+  # False by default.
+  expect_false(can_fetch_oauth_code("http://localhost:8080/redirect"))
+
+  # Only true in the presence of certain environment variables.
+  env <- c(
+    "HTTR2_OAUTH_CODE_SOURCE_URL" = "http://localhost:8080/code",
+    "HTTR2_OAUTH_REDIRECT_URL" = "http://localhost:8080/redirect"
+  )
+  withr::with_envvar(env, {
+    expect_true(can_fetch_oauth_code("http://localhost:8080/redirect"))
+
+    # Non-matching redirect URLs should not count as external sources, either.
+    expect_false(can_fetch_oauth_code("http://localhost:9090/redirect"))
+  })
+})
+
+# oauth_flow_auth_code_fetch ----------------------------------------------
+
+test_that("auth codes can be retrieved from an external source", {
+  skip_on_cran()
+
+  app <- webfakes::new_app()
```

Comment:
This test is a bit easier to write with webfakes (which we already use elsewhere)